bgpd/bgp_route.c

```diff
@@ -3205,7 +3200,6 @@ int bgp_update(struct peer *peer, struct prefix *p, uint32_t addpath_id,
 		pi_sub_type, &new_attr, rn)) {
 		peer->stat_pfx_nh_invalid++;
 		reason = "martian or self next-hop;";
-		bgp_attr_flush(&new_attr);
```

Comment:
This is incorrect, the `bgp_attr_add_gshut_community` call above in line 3193/3188 can change `attr->community` which needs to be freed here.